### DISCUSSION TITLE
SSR Loading Issue with Nuxt3 and Storyblok: Redundant API Calls Despite Payload File Generation

### DISCUSSION THREAD

**Discussion by jerome-barbato** [nuxt/nuxt#27835] [Questions]
Hi guys!

I've spent days on it and I still can't figure out why this happens. I've read the documentation 10 times, opened all the discussions about this, asked ChatGPT ^^ but it seems I'm the only one with this issue!

I'm using Nuxt3 with Storyblok, followed the [tutorial](https://www.storyblok.com/tp/add-a-headless-CMS-to-nuxt-3-in-5-minutes) and everything was working perfectly until I wanted to link my menu with a configuration story.

This is my setup:

node 18, nuxt 3.12.2, vue 3.4.30

In `layouts/default.vue`

```vue
<script setup>
const isPreview = useRuntimeConfig().public.NODE_ENV !== 'production'
const {locale} = useI18n()

const {data} = await useAsyncData( `${locale.value}-configuration`, async () => {
  const {data} = await useStoryblokApi().get(`cdn/stories/configuration`, {
    version: isPreview ? 'draft' : 'published',
    resolve_relations : 'chapters',
    language: locale.value
  })
  return data
})

onMounted(() => {

  if (isPreview && data.value && data.value.story.id)
    useStoryblokBridge(data.value.story.id, (evStory) => data.value.story = evStory)
})
</script>

<template>
  <div id="root">
    <Header :config="data.story.content" :rels="data.rels" v-if="data"/>
    <main>
      <slot/>
    </main>
    <Footer :config="data.story.content" :rels="data.rels" v-if="data"/>
  </div>
</template>
```

in `pages/[...slug].vue`

```vue
<script setup>

const {slug} = useRoute().params
const url = slug && slug.length > 0 ? slug.join('/') : 'home'

const isPreview = useRuntimeConfig().public.NODE_ENV !== 'production'
const publicUrl = useRuntimeConfig().public.url
const {locale} = useI18n()

const {data} = await useAsyncData(`${locale.value}-${url}`, async () => {
  const {data} = await useStoryblokApi().get(`cdn/stories/${url.replace(/\/$/, '')}`, {
    version: isPreview ? 'draft' : 'published',
    resolve_relations : 'prev,next',
    language: locale.value
  })
  return data
})

onMounted(() => {

  if (isPreview && data.value && data.value.story.id)
    useStoryblokBridge(data.value.story.id, (evStory) => data.value.story = evStory)
})
</script>

<template>
  <StoryblokComponent v-if="data" :blok="data.story.content" :story="data.story" :rels="data.rels"/>
</template>
```

Almost the exact same code but in a different place.

My issue: When building the SSR version, I have my HTML files hydrated, _payload.json generated, but while looking at the network requests, I still see a call to Storyblok to get the configuration story!

In the payload, I have both the page and configuration story.
I tried to move the code from `layouts/default.vue` to `components/header.vue` and `components/footer.vue`, same issue.

I'm out of ideas...

**Comment by maciekkus** [comment]
Hi @jerome-barbato 
Have you tried this:

```
const story = await useAsyncStoryblok(`${storySlug}`, { version: 'published' })
```

and only this, without useAsyncData

**Reply by jerome-barbato** [reply]
Hi @maciekkus, yes I tried, same effect
